openfisca_france/model/prestations/bourses_superieur.py

```diff
@@ -7,9 +13,247 @@ class echelon_bourse(Variable):
     label = "Echelon de la bourse perçue (de 0 à 7)"
     definition_period = MONTH
 
+    def formula(individu, period):
+        return individu('bourse_criteres_sociaux_echelon', period)
+
 
 class boursier(Variable):
     value_type = bool
     entity = Individu
     label = "Élève ou étudiant boursier"
     definition_period = MONTH
+
+    def formula(individu, period):
+        echelon = individu('bourse_criteres_sociaux_echelon', period)
+        return echelon >= 0
+
+
+class bourse_criteres_sociaux(Variable):
+    value_type = float
+    entity = Individu
+    reference = [
+        "Circulaire ESRS2013435C - Annexe 3 - Conditions de ressources et points de charge / 1 - Conditions de ressources"
+        ]
+    label = "Ressources prise en compte pour la bourse sur critères sociaux de l'enseignement supérieur"
+    definition_period = MONTH
+
+    def formula(individu, period, parameters):
+        montants = parameters(period).bourses_enseignement_superieur.criteres_sociaux.montants
+        echelon = individu('bourse_criteres_sociaux_echelon', period)
+
+        return montants.calc(echelon)
+
+
+class bourse_criteres_sociaux_eligibilite_etude(Variable):
+    value_type = bool
+    entity = Individu
+    reference = [
+        "Circulaire ESRS2013435C - Annexe 1 - Conditions d'études"
+        ]
+    label = "Satisfaction des critères d'âge pour la bourse sur critères sociaux de l'enseignement supérieur"
+    definition_period = MONTH
+
+    def formula(individu, period):
+        return individu('etudiant', period)
+
+
+class bourse_criteres_sociaux_eligibilite_age(Variable):
+    value_type = bool
+    entity = Individu
+    reference = [
+        "Circulaire ESRS2013435C - Annexe 2 - Critères d'attribution / 1 - Conditions d'âge"
+        ]
+    label = "Satisfaction des critères d'âge pour la bourse sur critères sociaux de l'enseignement supérieur"
+    definition_period = MONTH
+
+    def formula(individu, period):
+        age = individu('age', period)
+        handicap = individu('handicap', period)
+
+        return (age <= 28) + (handicap)
```

Comment:
N'avons-nous pas un risque d'afficher de fausses non éligibilités pour les étudiants ayant commencé à bénéficier d'une bourse et n'ayant pas interrompu leurs études ? :thinking:

> Pour mémo, le texte complet ([src](https://www.education.gouv.fr/bo/20/Hebdo25/ESRS2013435C.htm)) : 
Être âgé de moins de 28 ans au 1er septembre de l'année d'inscription dans une formation d'enseignement supérieur, dans le cas d'une première demande de bourse d'enseignement supérieur sur critères sociaux. À partir de 28 ans, l'étudiant ne doit pas interrompre ses études pour continuer à bénéficier d'une bourse.
La limite d'âge peut être reculée en fonction de la durée du service civique (articles L. 120-1 et suivants du Code du service national), du volontariat dans les armées (articles L. 121-1 et suivants du même Code) ou du volontariat international (articles L. 122-1 et suivants du même Code). Pour tout étudiant, la limite d'âge est reculée d'un an par enfant élevé.
Aucune limite d'âge n'est opposable à l'étudiant atteint d'un handicap reconnu par la Commission des droits et de l'autonomie des personnes handicapées.

Reply:
> Intégrer l'augmentation du plafond d'âge par nombre d'enfant élevé ?

Reply:
> Intégrer le cas d'exclusion "les personnes inscrites à Pôle Emploi comme demandeurs d'emploi" (non intuitif qu'on ne puisse pas demander une bourse et être en simple recherche d'emploi, non ?) ?

Reply:
Si si on a ce risque là mais je pense qu'on pourra itérer sur cette base là en fonction des retours des usagers/jeunes/experts.